ssi-sd-jwt/src/decode.rs

```diff
@@ -0,0 +1,221 @@
+use serde::de::DeserializeOwned;
+use serde::Deserialize;
+use ssi_jwk::JWK;
+use ssi_jwt::NumericDate;
+use std::collections::BTreeMap;
+
+use crate::serialized::deserialize_string_format;
+use crate::verify::{DecodedDisclosure, DisclosureKind};
+use crate::*;
+
+#[derive(Debug, Deserialize, PartialEq)]
+pub struct ValidityClaims {
+    pub nbf: Option<NumericDate>,
+    pub iat: Option<NumericDate>,
+    pub exp: Option<NumericDate>,
+}
+
+pub fn decode_verify_string_format<Claims: DeserializeOwned>(
+    serialized: &str,
+    key: &JWK,
+) -> Result<(ValidityClaims, Claims), Error> {
+    let deserialized =
+        deserialize_string_format(serialized).ok_or(Error::UnableToDeserializeStringFormat)?;
+
+    decode_verify(deserialized.jwt, key, &deserialized.disclosures)
+}
+
+pub fn decode_verify<Claims: DeserializeOwned>(
+    jwt: &str,
+    key: &JWK,
+    disclosures: &[&str],
```

Comment:
Do you think it could improve the API if this was using the `DecodedDisclosure` type? To force the user to "validate" the disclosures before doing anything, and also to prevent silly mistakes like passing the jwt as a disclosure